src/util/namespace.h

```diff
@@ -52,8 +52,20 @@ class namespace_baset
   const typet &follow_tag(const struct_tag_typet &src) const;
   const typet &follow_tag(const c_enum_tag_typet &src) const;
 
-  // these do the actual lookup
+  /// Returns the maximum integer x such that there is a symbol (in some of the
+  /// two symbol tables) whose name has \p prefix as prefix and x as suffix.
+  /// The intended use case is finding the next available symbol name for a
+  /// sequence of auto-generated symbols.
+  /// This method expects that all symbols whose name starts with \p prefix to
+  /// have a name of the form "prefixNNN" where NNN is an integer.
```

Comment:
No, this method does not have this expectation. It will work perfectly fine in absence of this property.